### DISCUSSION TITLE
Formatter Breaks `\author` Command with Optional Arguments in Elsevier Template

### DISCUSSION THREAD

**Discussion by edodeso** [overleaf-workshop/Overleaf-Workshop#210] [Q&A]
Hi,

I'm experiencing an issue with Overleaf Workshop where the formatter incorrectly modifies LaTeX commands that include optional arguments, particularly the `\author` command.

I am using the following document class:

`\documentclass[final, 5p, times, twocolumn]{elsarticle}`

Additionally, I am working with the Elsevier article template, available here [https://www.overleaf.com/latex/templates/elsevier-article-elsarticle-template/vdzfjgjbckgz].

For example, the following line:
`\author[label1]{John Doe}`
is reformatted as:
`\author{[}label1]{John Doe}`

This breaks the syntax and prevents successful compilation of the document.

**Steps to Reproduce**
Use Overleaf Workshop with formatting enabled in VS Code.
Add the Elsevier document class and the ` \author` command with an optional argument, as shown above.
Save the file or trigger the formatter.

**Expected Behaviour**
The formatter should preserve the syntax of the \author command, including its optional arguments, without modification.

**Actual Behaviour**
The formatter incorrectly splits or modifies the command, resulting in invalid syntax.

**Question**
Is there a way to configure the formatter in Overleaf Workshop to handle optional arguments in commands properly, or to skip formatting specific commands like `\author`?

Thank you for your help!

**Comment by edodeso** [comment]
@iamhyc Do you have any solution regarding this issue? Thanks!

**Reply by iamhyc** [reply]
Nope. Maybe @QianrenLi can help look at it.

**Reply by QianrenLi** [reply]
Since the formatter is basically using the one in [unified-latex](https://github.com/siefkenj/unified-latex) (by the way, great project), solution to this problem might be modifying the parsing rule. 
Skipping the specfic commands is a good point, however, I believe it is worth to take a bit of time to see how to fix this.
